app/controllers/user/payments_controller.rb

```diff
@@ -9,13 +9,19 @@ class PaymentsController < ::ApplicationController
 
       def index
         begin
-          customer = DiscourseSubscriptions::Customer.find_by(user_id: current_user.id, product_id: nil)
+          customer = Customer.find_by(user_id: current_user.id)
+          products = Product.all.pluck(:external_id)
 
           data = []
 
-          if customer.present?
+          if customer.present? && products.present?
+            # lots of matching because the Stripe API doesn't make it easy to match products => payments except from invoices
+            invoices = ::Stripe::Invoice.list(customer: customer[:customer_id])
+            invoices = invoices[:data].select { |invoice| products.include?(invoice[:lines][:data][0][:plan][:product]) }
+            invoices = invoices.map { |invoice| invoice[:id] }
```

Comment:
1. I think I would change variable names here to make it a little bit easier to follow, something like:
```ruby
 all_invoices = ::Stripe::Invoice.list(customer: customer[:customer_id])
 invoices_with_product = all_invoices[:data].select { |invoice| products.include?(invoice[:lines][:data][0][:plan][:product]) }
invoice_ids = invoices.map { |invoice| invoice[:id] })
```
2. `invoice[:lines][:data][0][:plan][:product]` - do we need `dig` here just in case something is nil on that route?
3. select and map can be combined into one loop using reduce but I am not sure - that may make code a little bit more efficient but also harder to read